include/envoy/secret/dynamic_secret_provider.h

```diff
@@ -0,0 +1,26 @@
+#pragma once
+
+#include <string>
+
+#include "envoy/ssl/tls_certificate_config.h"
+
+namespace Envoy {
+namespace Secret {
+
+/**
+ * An interface to fetch dynamic secret.
+ */
+class DynamicSecretProvider {
```

Comment:
Is the plan here to add other secret types? Otherwise should this be named DynamicTlsCertificateSecretProvider or similar?

Reply:
Yes, we plan to support other secret types, https://github.com/envoyproxy/data-plane-api/blob/master/envoy/api/v2/auth/cert.proto#L320. Is it okay to add TODO here and keep the name?

Reply:
This is a bit different from the static secret API, so I would match what we did there and make these typed, or change static to be similar.

Reply:
I see. Then I will rename it to DynamicTlsCertificateSecretProvider. Thanks!